test/run.d

```diff
@@ -157,51 +215,55 @@ auto predefinedTargets(string[] targets)
                 break;
 
             case "run_runnable_tests", "runnable":
-                newTargets.put(findFiles("runnable"));
+                newTargets.put(findFiles("runnable").map!createDefaultTarget);
                 break;
 
             case "run_fail_compilation_tests", "fail_compilation", "fail":
-                newTargets.put(findFiles("fail_compilation"));
+                newTargets.put(findFiles("fail_compilation").map!createDefaultTarget);
                 break;
 
             case "run_compilable_tests", "compilable", "compile":
-                newTargets.put(findFiles("compilable"));
+                newTargets.put(findFiles("compilable").map!createDefaultTarget);
                 break;
 
             case "all":
+                newTargets ~= createUnitTestTarget();
                 foreach (testDir; testDirs)
-                    newTargets.put(findFiles(testDir));
+                    newTargets.put(findFiles(testDir).map!createDefaultTarget);
+                break;
+            case "unit_tests":
+                newTargets ~= createUnitTestTarget();
                 break;
-
             default:
-                newTargets ~= t;
+                newTargets ~= createDefaultTarget(t);
         }
     }
     return newTargets.data;
 }
 
 // Removes targets that do not need updating (i.e. their .out file exists and is newer than the source file)
-auto filterTargets(string[] targets, string[string] env)
+auto filterTargets(Target[] targets, string[string] env)
 {
     bool error;
     foreach (target; targets)
     {
-        if (!testPath(target).exists)
+        if (!target.exists)
         {
-            writefln("Warning: %s can't be found", target);
+            writefln("Warning: %s can't be found", target.normalizedTestName);
```

Comment:
What's wrong with the status quo?

```
Warning: compilable/test12009i.d can't be found
```

Reply:
There's no change, see: https://github.com/dlang/dmd/pull/8528#discussion_r25398297.7